rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1721,13 +1721,24 @@ public RMInitialState getRMInitialState() {
             }
         }
 
-        final Map<String, RMNodeSourceEvent> nodeSourceEvents = this.definedNodeSources.values()
-                                                                                       .stream()
-                                                                                       .map(NodeSource::createNodeSourceEvent)
-                                                                                       .collect(Collectors.toMap(RMNodeSourceEvent::getSourceName,
-                                                                                                                 event -> event));
+        final List<RMNodeSourceEvent> nodeSourceEvents = new ArrayList<>(this.definedNodeSources.values()
+                                                                                                .stream()
+                                                                                                .map(NodeSource::createNodeSourceEvent)
+                                                                                                .collect(Collectors.toList()));
 
-        return new RMInitialState(nodeEvents, nodeSourceEvents);
+        long eventCounter = 0;
+        for (RMNodeSourceEvent nodeSourceEvent : nodeSourceEvents) {
+            nodeSourceEvent.setCounter(eventCounter++);
+        }
+        for (RMNodeEvent nodeEvent : nodeEvents.values()) {
+            nodeEvent.setCounter(eventCounter++);
+        }
```

Comment:
Replace by 
nodeSourceEvents.forEach(nodeSourceEvent - > nodeSourceEvent.setCounter(eventCounter++));
nodeEvents.values().forEach(nodeEvent -> nodeEvent.setCounter(eventCounter++));
?

Reply:
if eventCounter just long, and it is long, then i can change it from inside of lambda.